src/driver/sqlserver/SqlServerQueryRunner.ts

```diff
@@ -1505,126 +1510,165 @@ export class SqlServerQueryRunner extends BaseQueryRunner implements QueryRunner
+            const tablesSql = dbNames.map(dbName => {
+                return `
+                    SELECT
+                        "TABLE_CATALOG", "TABLE_SCHEMA", "TABLE_NAME"
+                    FROM "${dbName}"."INFORMATION_SCHEMA"."TABLES"
+                    WHERE "TABLE_TYPE" = 'BASE TABLE' AND ${tablesCondition}`;
+            }).join(" UNION ALL ");
 
-        // load tables, columns, indices and foreign keys
-        const schemaNamesString = schemaNames.map(name => "'" + name + "'").join(", ");
+            dbTables.push(...await this.query(tablesSql));
+        }
 
-        const tablesCondition = tableNames.map(tableName => {
-            const [schema, name] = extractTableSchemaAndName(tableName);
-            return `("TABLE_SCHEMA" = '${schema}' AND "TABLE_NAME" = '${name}')`;
-        }).join(" OR ");
+        // if tables were not found in the db, no need to proceed
+        if (dbTables.length === 0) {
+            return [];
+        }
 
-        const tablesSql = dbNames.map(dbName => {
-            return `SELECT * FROM "${dbName}"."INFORMATION_SCHEMA"."TABLES" WHERE ` + tablesCondition;
-        }).join(" UNION ALL ");
+        const dbTablesByCatalog = dbTables.reduce((c, { TABLE_CATALOG, ...other }) => {
+            c[TABLE_CATALOG] = c[TABLE_CATALOG] || [];
+            c[TABLE_CATALOG].push(other);
+            return c;
+        }, {} as { [key: string ]: { TABLE_NAME: string, TABLE_SCHEMA: string }[] })
+
+        const columnsSql = Object.entries(dbTablesByCatalog).map(([ TABLE_CATALOG, tables ]) => {
+            const condition = tables.map(
+                ({ TABLE_SCHEMA, TABLE_NAME }) => `("TABLE_SCHEMA" = '${TABLE_SCHEMA}' AND "TABLE_NAME" = '${TABLE_NAME}')`
+            ).join("OR");
 
-        const columnsSql = dbNames.map(dbName => {
-            return `SELECT * FROM "${dbName}"."INFORMATION_SCHEMA"."COLUMNS" WHERE ` + tablesCondition;
+            return `SELECT * FROM "${TABLE_CATALOG}"."INFORMATION_SCHEMA"."COLUMNS" WHERE (${condition})`;
         }).join(" UNION ALL ");
 
-        const constraintsCondition = tableNames.map(tableName => {
-            const [schema, name] = extractTableSchemaAndName(tableName);
-            return `("columnUsages"."TABLE_SCHEMA" = '${schema}' AND "columnUsages"."TABLE_NAME" = '${name}' ` +
-             `AND "tableConstraints"."TABLE_SCHEMA" = '${schema}' AND "tableConstraints"."TABLE_NAME" = '${name}')`;
-        }).join(" OR ");
+        const constraintsSql = Object.entries(dbTablesByCatalog).map(([ TABLE_CATALOG, tables ]) => {
+            const conditions = tables.map(({ TABLE_NAME, TABLE_SCHEMA }) => {
+                return (
+                    `(` +
+                    `"columnUsages"."TABLE_SCHEMA" = '${TABLE_SCHEMA}' AND "columnUsages"."TABLE_NAME" = '${TABLE_NAME}' AND ` +
+                    `"tableConstraints"."TABLE_SCHEMA" = '${TABLE_SCHEMA}' AND "tableConstraints"."TABLE_NAME" = '${TABLE_NAME}'` +
```

Comment:
Move this into the join instead to simplify this condition and cut down on parameters?